pyof/v0x01/controller2switch/packet_out.py

```diff
@@ -112,13 +112,17 @@ def _update_actions_len(self):
             self.actions_len = ListOfActions(self.actions).get_size()
 
     def _validate_in_port(self):
-        port = self.in_port
         valid = True
-        if isinstance(port, Port):
-            if port not in _VIRT_IN_PORTS:
+        if (isinstance(self.in_port, int) and
+                (self.in_port > 0 and self.in_port < Port.OFPP_MAX.value)):
```

Comment:
Port is a IntEnum, thus you do not need to use the `.value` here, it's elements can be directly compared to integers out of the box.
Also, why comparing if the `in_port` is **less then** `Port.OFPP_MAX` instead of **less or equal to** ?

```python
In [4]: Port.OFPP_MAX
Out[4]: <Port.OFPP_MAX: 65280>

In [5]: 65280 == Port.OFPP_MAX
Out[5]: True

In [6]: 65280 >= Port.OFPP_MAX
Out[6]: True

In [7]: 65280 > Port.OFPP_MAX
Out[7]: False

In [8]: 65280 < Port.OFPP_MAX
Out[8]: False

In [9]: 65281 <= Port.OFPP_MAX
Out[9]: False

In [10]: 65279 <= Port.OFPP_MAX
Out[10]: True
```